src/aiida_quantumespresso/tools/calculations/pw.py

```diff
@@ -51,3 +53,48 @@ def get_scf_accuracy(self, index=0):
             return scf_accuracy[scf_accuracy_index[index - 1]:scf_accuracy_index[index]]
 
         return scf_accuracy[scf_accuracy_index[index]:scf_accuracy_index[index + 1]]
+
+    def get_magnetic_configuration(self, atol: float = 0.5, ztol: float = 0.05) -> AttributeDict:
+        """Get the final magnetic configuration of a ``pw.x`` calculation."""
+
+        calculation = self._node.inputs.parameters.get_dict()['CONTROL'].get('calculation', 'scf')
+
+        if calculation in ['scf', 'nscf', 'bands']:
+            structure = self._node.inputs.structure
+        else:
+            structure = self._node.outputs.output_structure
```

Comment:
Think here it might be better to just always try to get the `output_structure` and catch the exception if it doesn't exist and fallback on the input